### PR TITLE
Specify minimum version of sphinx_rtd_theme to allow most recent release of docutils

### PR DESCRIPTION
**Pull request by namurphy**
Earlier this year we had some difficulties with docutils 0.17 and higher, which appeared to be incompatibilities with sphinx_rtd_theme.  I checked the changelog for sphinx_rtd_theme, and the recently released version 1.0.0 is apparently now compatible with more recent versions of docutils.  This PR updates the requirements for both docutils and sphinx_rtd_theme to test that this fix actually does work.

### CODE REVIEW

**Comment by namurphy** [comment]
Fortunately, this seems to be working!  I'm not sure about which minimum version of docutils should be allowed, so I'll need to:

 - [ ] Determine what the minimum version of docutils should be

**Comment by StanczakDominik** [comment]
> There is a minimum version of docutils required by Sphinx, so I don't think we need to specify one ourselves.

I would agree if we were only depending on functionality provided by Sphinx itself; but `plasmapy-sphinx` makes me less sure of that!

**Comment by namurphy** [comment]
The two problems with docutils ≥ 0.17 that I knew of were...
 - Odd padding of the sidebar #docutils (#1107)
 - Bullet points not showing up in lists (#1230)
 
Both of those seem to no longer be problems in the docs preview with sphinx_rtd_theme ≥ 1.0.0 (looking at doc previews on Firefox from Ubuntu).  @rocco8773 — is there any reason why we'd need to set a minimum version for docutils?

File: changelog/1275.doc.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Updated the requirements for the documentation build to be docutils ≥ 0.17.1
+and sphinx_rtd_them ≥ 1.0.0.
```
Comment (rocco8773) [inline changelog/1275.doc.rst:2]:
```suggestion
Updated the requirements for the documentation build to be no restrictions on `docutils`
and `sphinx_rtd_them >= 1.0.0`.  `docutils == 0.17` is not compatible with
`sphinx_rtd_theme < 1.0` (see :pr:`1107`) and `docutils == 0.17` is not compatible with
`sphinx_rtd_theme < 1.0` (see :issue:`1230`).
```

**Comment by namurphy** [comment]
- [ ] Check that `:issue:` and `:pr:` work in changelog entry

**Review by rocco8773 — APPROVED** [review]

**Pull request merged**